src/main/java/org/prebid/server/auction/AmpRequestFactory.java

```diff
@@ -221,37 +227,64 @@ private BidRequest overrideParameters(BidRequest bidRequest, HttpServerRequest r
             }
         }
 
-        final Site updatedSite = overrideSite(bidRequest.getSite(), request);
-        final Imp updatedImp = overrideImp(bidRequest.getImp().get(0), request);
+        final String requestTargeting = request.getParam(TARGETING_REQUEST_PARAM);
+        final Targeting targeting = parseTargeting(requestTargeting);
+        final ObjectNode targetingNode = (ObjectNode) readTargeting(requestTargeting);
```

Comment:
This cast is worrisome - why not check if parsed `JsonNode` is object and make cast in `readTargeting`?

Reply:
fixed